packages/client/hmi-client/src/components/workflow/ops/compare-datasets/compare-datasets-utils.ts

```diff
@@ -269,28 +270,42 @@ export function generateRankingCharts(
 		allRankedCriteriaValues.push(sortedRankingCriteriaValues);
 	});
 
-	// Sum up the values of the same intervention policy
-	const valueMap: Record<string, number> = {};
-	allRankedCriteriaValues.flat().forEach(({ score, policyName }) => {
-		if (valueMap[policyName]) {
-			valueMap[policyName] += score;
-		} else {
-			valueMap[policyName] = score;
-		}
+	// For each criteria
+	allRankedCriteriaValues.forEach((criteriaValues, index) => {
+		const rankMutliplier = node.state.criteriaOfInterestCards[index].rank === RankOption.MINIMUM ? -1 : 1;
+
+		// Calculate mean and stdev for this criteria
+		const meanValue = criteriaValues.reduce((acc, val) => acc + val.score, 0) / criteriaValues.length;
+		let stdevValue = Math.sqrt(
+			criteriaValues.reduce((acc, val) => acc + (val.score - meanValue) ** 2, 0) / (criteriaValues.length - 1)
+		);
+		if (stdevValue === 0) stdevValue = 1;
```

Comment:
We have some of these common metric calculations like stddev under `utils/stats.ts` - can you see if we can reuse them rather than making separate copies?

Reply:
thanks